snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -532,6 +533,11 @@ def _compose_snap_yaml(self):
 
         self._process_passthrough_properties(snap_yaml)
 
+        # FIXME: Remove this warning once layouts are stable and no longer behind a
+        # feature flag in snapd.
```

Comment:
is there a way to dynamically check this?

Reply:
We chatted about this in IRC. `snap pack --check-skeleton` will catch invalid layouts, so @sergiusens requested that we not check the contents of the layout at all and just pass it through to the snap.yaml. This means we can't catch mistakes before the snap is built, but it gives snapd the freedom to change without requiring changes in snapcraft. I'll propose a new PR for this.